### PR TITLE
Rest essentials draft

### PR DESCRIPTION
**Pull request by KerstenBreuer**
Draft of all of the essential endpoints for a typical user walkthrough as detailed in #90.

Please approve the overall design.

Please don't be that critical while reviewing the actual Response/Request schemas, as they are just meant as a starting point and will likely change when we are actually implementing the backend logic.

### CODE REVIEW

File: datameta/api/openapi.yaml [hunk 1]
```diff
@@ -163,6 +473,166 @@ components:
...
+        
+
+    FileAnnouncement:
+      type: object
+      properties:
+        name:
+          type: string
+        checksum:
+          type: string
+        userId:
```
Comment (lkuchenb) [inline datameta/api/openapi.yaml:537]: Is it necessary to capture the user ID here?

File: datameta/api/openapi.yaml [hunk 2]
```diff
@@ -163,6 +473,166 @@ components:
...
+        # in the backend
+      required:
+        - name
+        - checksum
+        - userId
+      additionalProperties: false
+    
+    FileUploadResponse:
+      type: object
+      properties:
```
Comment (lkuchenb) [inline datameta/api/openapi.yaml:549]: We may additionally need expiry information

File: datameta/api/openapi.yaml [hunk 3]
```diff
@@ -163,6 +473,166 @@ components:
...
+    FileResponse:
+      type: object
+      properties:
+        name:
+          type: string
+        fileId:
+          type: string
+        contentUploaded:
+          type: boolean
+          nullable: true
```
Comment (lkuchenb) [inline datameta/api/openapi.yaml:577]: In which case do we need this to be `null`?

**Review by lkuchenb — CHANGES_REQUESTED** [review]
Looks generally good to me, I've left a few comments and the compose file shouldn't be in the PR

**Comment by lkuchenb** [comment]
I missed that the compose yml is not the main one but the devcontainer one 👍 

I'll merge this, let's keep track of the comments I made, I'd like to merge another PR that requires DB migration and those would be hard to merge.

File: .devcontainer/docker-compose.yml [hunk 4]
```diff
@@ -25,6 +25,23 @@ services:
...
+      DATAMETA_INITIAL_FULLNAME: "admin"
+      DATAMETA_INITIAL_EMAIL: "admin@admin.admin"
+      DATAMETA_INITIAL_PASS: "admin"
+      DATAMETA_INITIAL_GROUPNAME: "admin"
+      WAITRESS_MAX_REQUEST_BODY_SIZE: 10737418240 # 10 GB
+      WAITRESS_LISTEN: "*:8080"
+      WAITRESS_URL_SCHEME:
+      WAITRESS_TRUSTED_PROXY_HEADERS:
+      WAITRESS_TRUSTED_PROXY:
+
```
[the remark this answers is not in the document]
Comment (KerstenBreuer) [reply .devcontainer/docker-compose.yml:44]: Yes, as you mentioned below, this is just used by vscode

File: datameta/api/openapi.yaml [hunk 5]
```diff
@@ -64,6 +66,45 @@ paths:
...
+          If your group already exists, please refer to it using
+          its "groupId" (not name).
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/RegRequest"
+        '403':
```
[the remark this answers is not in the document]
Comment (KerstenBreuer) [reply datameta/api/openapi.yaml:101]: sorry, which line are you refering to?

File: datameta/api/openapi.yaml (hunk 1 above)
Comment (KerstenBreuer) [reply datameta/api/openapi.yaml:537]: No, actually not, I'll remove it

File: datameta/api/openapi.yaml (hunk 2 above)
Comment (KerstenBreuer) [reply datameta/api/openapi.yaml:549]: done, also included that in the db model

File: datameta/api/openapi.yaml (hunk 1 above)
Comment (KerstenBreuer) [reply datameta/api/openapi.yaml:537]: Done

File: datameta/api/openapi.yaml (hunk 3 above)
Comment (KerstenBreuer) [reply datameta/api/openapi.yaml:577]: No case, will be removed

File: datameta/api/openapi.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (lkuchenb) [reply datameta/api/openapi.yaml:101]: The 403 response

**Pull request merged**